### PR TITLE
Update to eslint flat config and yea/nay all rules

### PR DESCRIPTION
**Pull request by reosarevok**
eslint 9 is now out: https://eslint.org/blog/2024/04/eslint-v9.0.0-released/

There's a new configuration default, "flat config" (https://eslint.org/docs/latest/use/configure/configuration-files) - this is supported in eslint 8, default in eslint 9, and will be the only option in eslint 10. The first commits here move our config to use this system. While doing this I changed the way we deal with unfixed rules, which now are considered as fixed and enforced globally while disabling them *only* where they are known to still fail. This should help avoid introducing new code that doesn't follow the rules.

Stylistic rules are deprecated in the main eslint package because they have been moved to `@stylistic/eslint-plugin-js`, so this moves them to use the new package.

Actually moving to eslint 9 cannot happen at least until `eslint-plugin-react` fixes its compatibility (unless we want to just comment out all their rules temporarily, but I don't think that'd be too smart), but at least we can migrate the config, which only conflicts with a couple rules that need updating (and are commented out for now in the cases where we will probably want them later).

Further commits look through all the rules from the plugins we use, implement all that seem useful, and explicitly turn off all others, to make it clear that we have indeed looked into them and do not want them.

### CODE REVIEW

**Comment by reosarevok** [comment]
@brainzbot, retest this please

**Comment by reosarevok** [comment]
@brainzbot, retest this please

File: root/static/lib/flot/jquery.flot.js [hunk 1]
```diff
@@ -34,10 +34,6 @@ Licensed under the MIT license.
 // the actual Flot code
 (function($) {
 
-	// Cache the prototype hasOwnProperty for faster access
-
-	var hasOwnProperty = Object.prototype.hasOwnProperty;
```
Comment (mwiencek) [inline root/static/lib/flot/jquery.flot.js:39]: Does ESLint actually complain about files under root/static/lib/? (I would avoid touching these in general, not that the changes seem problematic.)

File: root/static/scripts/edit/URLCleanup.js [hunk 2]
```diff
@@ -2598,13 +2502,10 @@ const CLEANUPS: CleanupEntries = {
...
        */
-      url = url.replace(new RegExp(
-        '([&?])(__tn__|_fb_noscript|_rdr|acontext|em|entry_point|filter|' +
-        'focus_composer|fref|hc_location|pnref|qsefr|ref|' +
-        'ref_dashboard_filter|ref_page_id|ref_type|refsrc|rf|' +
-        'sid_reminder|sk|tab|viewas)=([^?&]*)',
-        'g',
-      ), '$1');
+      url = url.replace(
+        /([&?])(__tn__|_fb_noscript|_rdr|acontext|em|entry_point|filter|focus_composer|fref|hc_location|pnref|qsefr|ref|ref_dashboard_filter|ref_page_id|ref_type|refsrc|rf|sid_reminder|sk|tab|viewas)=([^?&]*)/g,
```
Comment (mwiencek) [inline root/static/scripts/edit/URLCleanup.js:2506]: Seems like an improvement in the majority of cases, but maybe exceptions can be added for cases like this, where having it across multiple lines is easier to read, and we aren't benefiting from reduced escaping.

File: eslint.config.mjs [hunk 3]
```diff
@@ -465,7 +465,10 @@ export default [
       }],
       'eslint-plugin-js/dot-location': ['warn', 'property'],
       'eslint-plugin-js/eol-last': ['warn', 'always'],
-      'eslint-plugin-js/func-call-spacing': ['warn', 'never'],
+      'eslint-plugin-js/function-call-argument-newline': [
```
Comment (mwiencek) [inline eslint.config.mjs:468]: Most of the changes in this commit don't seem more readable to me, TBH.

Also, I don't see how

```js
foo('bar',
    'baz);
```

is much different from

```js
foo={'bar' +
     'baz'}
```

which is allowed by our JSX rules?

**Review by mwiencek — CHANGES_REQUESTED** [review]
Thanks for all your hard work on this! Looks to be almost ready, but there is a new conflict and a test failure (which I haven't looked at).

File: root/static/lib/flot/jquery.flot.js (hunk 1 above)
Comment (reosarevok) [reply root/static/lib/flot/jquery.flot.js:39]: I don't remember if it complains, IIRC we just decided that `hasOwn` seemed better so I made the change everywhere...

File: root/static/scripts/edit/URLCleanup.js (hunk 2 above)
Comment (reosarevok) [reply root/static/scripts/edit/URLCleanup.js:2506]: Turns out this one isn't even caught by the rule so we don't need an exception.

**Comment by reosarevok** [comment]
@brainzbot, retest this please

File: eslint.config.mjs (hunk 3 above)
Comment (mwiencek) [reply eslint.config.mjs]: I guess the only real case where I liked them on the same line are the mutate-cow property paths. Though they're not that bad on separate lines either, so I guess the rule is worth having overall.

**Review by mwiencek — APPROVED** [review]

**Pull request merged**